### PR TITLE
adding more info and examples to docs + upgrade to latest flake8 checks

### PR DESCRIPTION
**Pull request by theweaklink**

### CODE REVIEW

File: onecode/elements/output/csv_output.py [hunk 1]
```diff
@@ -73,9 +79,9 @@ def __init__(
     def value(self) -> str:
         """
         Returns:
-            The path or list of paths for the output file(s): if paths are not absolute, then
-            they are considered relative to the data output folder. See
-            [Best Practices With Data][best-practices-with-data] for more information.
+            The path to the output file: if path are not absolute, then it is considered relative
```
Comment (hameye) [inline onecode/elements/output/csv_output.py:82]: List of path are not supported anymore ? Or was it a typo ?

File: onecode/elements/output/csv_output.py [hunk 2]
```diff
@@ -44,16 +44,22 @@ def __init__(
 
         !!! example
             ```py
+            import pandas as pd
             from onecode import csv_output, Mode, Project
 
-            Project().mode = Mode.CONSOLE
-            widget = csv_output(
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
```
Comment (hameye) [inline onecode/elements/output/csv_output.py:51]:
```suggestion
            Project().current_flow = "test"
```

File: onecode/elements/output/csv_output.py [hunk 3]
```diff
@@ -44,16 +44,22 @@ def __init__(
...
+
+            csv_file = csv_output(
                 key="CsvOutput",
                 value="/path/to/file.csv",
                 label="My CsvOutput",
                 tags=['CSV']
             )
-            print(widget)
+
+            df = pd.DataFrame([[0, 1, 2], [3, 4, 5]], columns=['A', 'B', 'C'])
```
Comment (hameye) [inline onecode/elements/output/csv_output.py:60]:
```suggestion
            df = pd.DataFrame([[0, 1, 2], [3, 4, 5]], columns=["A", "B", "C"])
```

File: onecode/elements/output/csv_output.py [hunk 4]
```diff
@@ -44,16 +44,22 @@ def __init__(
...
-            Project().mode = Mode.CONSOLE
-            widget = csv_output(
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
+
+            csv_file = csv_output(
                 key="CsvOutput",
                 value="/path/to/file.csv",
                 label="My CsvOutput",
                 tags=['CSV']
```
Comment (hameye) [inline onecode/elements/output/csv_output.py:57]:
```suggestion
                tags=["CSV"]
```

File: onecode/elements/output/file_output.py [hunk 5]
```diff
@@ -45,14 +45,20 @@ def __init__(
             ```py
             from onecode import file_output, Mode, Project
 
-            Project().mode = Mode.CONSOLE
-            widget = file_output(
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
```
Comment (hameye) [inline onecode/elements/output/file_output.py:49]:
```suggestion
            Project().current_flow = "test"
```

File: onecode/elements/output/file_output.py [hunk 6]
```diff
@@ -45,14 +45,20 @@ def __init__(
...
+
+            file = file_output(
                 key="FileOutput",
                 value="/path/to/file.txt",
                 label="My FileOutput",
                 tags=['TXT']
```
Comment (hameye) [inline onecode/elements/output/file_output.py:55]:
```suggestion
                tags=["TXT"]
```

File: onecode/elements/output/file_output.py [hunk 7]
```diff
@@ -45,14 +45,20 @@ def __init__(
...
             )
-            print(widget)
+
+            with open(file, 'w') as f:
+                f.write('Hello OneCode!')
```
Comment (hameye) [inline onecode/elements/output/file_output.py:59]:
```suggestion
            with open(file, "w") as f:
                f.write("Hello OneCode!")
```

File: onecode/elements/output/image_output.py [hunk 8]
```diff
@@ -45,16 +45,29 @@ def __init__(
...
         !!! example
             ```py
+            import matplotlib.pyplot as plt
+            import numpy as np
             from onecode import image_output, Mode, Project
 
-            Project().mode = Mode.CONSOLE
-            widget = image_output(
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
```
Comment (hameye) [inline onecode/elements/output/image_output.py:53]:
```suggestion
            Project().current_flow = "test"
```

File: onecode/elements/output/image_output.py [hunk 9]
```diff
@@ -45,16 +45,29 @@ def __init__(
...
+
+            image_file = image_output(
                 key="ImageOutput",
                 value="/path/to/file.jpg",
                 label="My ImageOutput",
                 tags=['Image']
```
Comment (hameye) [inline onecode/elements/output/image_output.py:59]:
```suggestion
                tags=["Image"]
```

File: onecode/elements/output/image_output.py [hunk 10]
```diff
@@ -45,16 +45,29 @@ def __init__(
...
             )
-            print(widget)
+
+            t = np.arange(0.0, 2.0, 0.01)
```
Comment (hameye) [inline onecode/elements/output/image_output.py:62]: Use parameters name in function call for clarity ?

File: onecode/elements/output/image_output.py [hunk 11]
```diff
@@ -45,16 +45,29 @@ def __init__(
...
+            s = 1 + np.sin(2 * np.pi * t)
+            fig, ax = plt.subplots()
+            ax.plot(t, s)
+            ax.set(xlabel='time (s)', ylabel='voltage (mV)')
```
Comment (hameye) [inline onecode/elements/output/image_output.py:66]:
```suggestion
            ax.set(xlabel="time (s)", ylabel="voltage (mV)")
```

File: onecode/elements/output/plotly_output.py [hunk 12]
```diff
@@ -44,16 +47,28 @@ def __init__(
 
         !!! example
             ```py
-            from onecode import csv_output, Mode, Project
+            import plotly
+            import plotly.graph_objects as go
+            from onecode import plotly_output, Mode, Project
+
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
```
Comment (hameye) [inline onecode/elements/output/plotly_output.py:55]:
```suggestion
            Project().current_flow = "test"
```

File: onecode/elements/output/plotly_output.py [hunk 13]
```diff
@@ -44,16 +47,28 @@ def __init__(
...
 
-            Project().mode = Mode.CONSOLE
-            widget = plotly_output(
+            plot_file = plotly_output(
                 key="PlotlyOutput",
                 value="/path/to/file.json",
                 label="My PlotlyOutput",
                 tags=['Graph']
```
Comment (hameye) [inline onecode/elements/output/plotly_output.py:61]:
```suggestion
                tags=["Graph"]
```

File: onecode/elements/output/text_output.py [hunk 14]
```diff
@@ -47,14 +47,19 @@ def __init__(
             ```py
             from onecode import text_output, Mode, Project
 
-            Project().mode = Mode.CONSOLE
-            widget = text_output(
+            Project().mode = Mode.EXECUTE
+            Project().current_flow = 'test'
```
Comment (hameye) [inline onecode/elements/output/text_output.py:51]:
```suggestion
            Project().current_flow = "test"
```

File: onecode/elements/output/text_output.py [hunk 15]
```diff
@@ -47,14 +47,19 @@ def __init__(
...
+
+            text_file = text_output(
                 key="TextOutput",
                 value="/path/to/file.txt",
                 label="My TextOutput",
                 tags=['Text'],
```
Comment (hameye) [inline onecode/elements/output/text_output.py:57]:
```suggestion
                tags=["Text"],
```

File: onecode/elements/output/text_output.py [hunk 16]
```diff
@@ -47,14 +47,19 @@ def __init__(
...
                 truncate_at=1000
             )
+
+            with open(text_file, 'w') as f:
+                f.write('Hello OneCode!')
```
Comment (hameye) [inline onecode/elements/output/text_output.py:62]:
```suggestion
            with open(text_file, "w") as f:
                f.write("Hello OneCode!")
```

**Review by hameye — CHANGES_REQUESTED** [review]

File: onecode/elements/output/csv_output.py (hunk 1 above)
Comment (theweaklink) [reply onecode/elements/output/csv_output.py:82]: It was an erroneous description (probably from a copy/paste of the input elements), it cannot be a list of path

**Review by hameye — APPROVED** [review]

**Pull request merged**